src/ui/UserInterface.java

```diff
@@ -311,20 +312,28 @@ public void enrollMember() {
 	 */
 	public void removeMember() {
 		Request.instance().setMemberId(getString("\nEnter member's ID to be removed: "));
-		Result result = groceryStore.removeMember(Request.instance());
-		switch (result.getResultCode()) {
-		case Result.INVALID_MEMBER_ID:
-			System.out
-					.println("No such member with id: " + Request.instance().getMemberId() + ", at the grocery store.");
-			break;
-		case Result.ACTION_FAILED:
-			System.out.println("Member could not be removed.");
-			break;
-		case Result.ACTION_SUCCESSFUL:
-			System.out.println("Member has been removed.");
-			break;
-		default:
-			System.out.println("An error has occured.");
+		if (groceryStore.memberIdExists(Request.instance().getMemberId())) {
+			Result result = groceryStore.getMember(Request.instance().getMemberId());
+			System.out.println("You are about to remove member " + result.getMemberId() + ", " + result.getMemberName()
+					+ ", ph. number " + result.getMemberPhoneNumber() + ", from the system.");
+			if (getYesOrNo("Are you sure?")) {
+				result = groceryStore.removeMember(Request.instance());
+				switch (result.getResultCode()) {
+				case Result.ACTION_FAILED:
+					System.out.println("Member could not be removed.");
+					break;
+				case Result.ACTION_SUCCESSFUL:
+					System.out.println("Member " + result.getMemberId() + " has been removed.");
+					break;
+				default:
+					System.out.println("An error has occured.");
+				}
+			} else {
+				System.out.println("Member " + result.getMemberId() + " will not be removed.");
+			}
+		} else {
+			System.out.println("No such member with id: " + Request.instance().getMemberId().toUpperCase()
+					+ ", at the grocery store.");
```

Comment:
So the main change in this PR is that the UI checks for the member and also asks the user to confirm.

Is that correct? @paveldanek

Reply:
Yes, exactly.